unit-tests/src/test/scala/re2s/RE2PatternSuite.scala

```diff
@@ -0,0 +1,210 @@
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           Array[String]("a", "b", "cc", "", "d", "e"))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           3,
+                           Array[String]("a", "b", "cc//d/e//"))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           4,
+                           Array[String]("a", "b", "cc", "/d/e//"))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           5,
+                           Array[String]("a", "b", "cc", "", "d/e//"))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           6,
+                           Array[String]("a", "b", "cc", "", "d", "e//"))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           7,
+                           Array[String]("a", "b", "cc", "", "d", "e", "/"))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           8,
+                           Array[String]("a", "b", "cc", "", "d", "e", "", ""))
+    ApiTestUtils.testSplit("/",
+                           "a/b/cc//d/e//",
+                           9,
+                           Array[String]("a", "b", "cc", "", "d", "e", "", ""))
+    // The tests below are listed at
+    // http://docs.oracle.com/javase/1.5.0/docs/api/java/util/regex/Pattern.html#split(java.lang.CharSequence, int)
+    val s       = "boo:and:foo"
+    val regexp1 = ":"
+    val regexp2 = "o"
+    ApiTestUtils.testSplit(regexp1, s, 2, Array[String]("boo", "and:foo"))
+    ApiTestUtils.testSplit(regexp1, s, 5, Array[String]("boo", "and", "foo"))
+    ApiTestUtils.testSplit(regexp1, s, -2, Array[String]("boo", "and", "foo"))
+    ApiTestUtils.testSplit(regexp2,
+                           s,
+                           5,
+                           Array[String]("b", "", ":and:f", "", ""))
+    ApiTestUtils.testSplit(regexp2,
+                           s,
+                           -2,
+                           Array[String]("b", "", ":and:f", "", ""))
+    ApiTestUtils.testSplit(regexp2, s, 0, Array[String]("b", "", ":and:f"))
+    ApiTestUtils.testSplit(regexp2, s, Array[String]("b", "", ":and:f"))
+  }
+
+  test("GroupCount") { // It is a simple delegation, but still test it.
+    ApiTestUtils.testGroupCount("(.*)ab(.*)a", 2)
+    ApiTestUtils.testGroupCount("(.*)(ab)(.*)a", 3)
+    ApiTestUtils.testGroupCount("(.*)((a)b)(.*)a", 4)
+    ApiTestUtils.testGroupCount("(.*)(\\(ab)(.*)a", 3)
+    ApiTestUtils.testGroupCount("(.*)(\\(a\\)b)(.*)a", 3)
+  }
+
+  test("Quote") {
+    ApiTestUtils.testMatchesRE2(Pattern.quote("ab+c"), 0, "ab+c", "abc")
+  }
+
+  private def reserialize(obj: Pattern) = {
+
+    obj // Remove when Object*Stream methods are implemented
```

Comment:
We don't really plan to support `Object*Stream` (i.e., java serialization), so might as well just drop code that depends on it.

Reply:
Changed as recommended.  Thank you for the improvement & simplification.  Less lore to
remember or re-create.

Passes Travis CI.